WooCommerce/Classes/ViewRelated/Products/Edit Product/EditCategories/ProductCategoryListViewController.swift

```diff
@@ -67,8 +68,18 @@ private extension ProductCategoryListViewController {
 private extension ProductCategoryListViewController {
     func configureViewModel() {
         viewModel.performInitialFetch()
-        viewModel.observeCategoryListChanges { [weak self] in
-            self?.tableView.reloadData()
+        viewModel.observeCategoryListStateChanges { [weak self] syncState in
+            switch syncState {
+            case .syncing:
+                self?.displayPlaceholderCategories()
+            case let .failed(pageNumber):
+                self?.removePlaceholderCategories()
+                self?.displaySyncingErrorNotice(pageNumber: pageNumber)
+            case .synced:
+                self?.removePlaceholderCategories()
+            default:
```

Comment:
Should we use `.initialized` instead? I think that's better in case we add more options in the future. 🤔 If we add more options, `default` would catch them and we might forget to add more branches here.

Reply:
https://github.com/woocommerce/woocommerce-ios/pull/2130/commits/d35665e4ca1bac21c5097ea604be5b5990d5d739